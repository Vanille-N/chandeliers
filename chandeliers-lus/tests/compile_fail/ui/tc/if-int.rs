chandeliers_lus::decl! {
    node foo() returns ();
    let
        assert if 1 then true else false;
    tel;
}

chandeliers_lus::decl! {
    node foo() returns ();
    let
        assert if (1, 2, 3) then true else false;
    tel;
}

chandeliers_lus::decl! {
    node foo() returns ();
    let
        assert 1;
    tel;
}

chandeliers_lus::decl! {
    node foo() returns ();
    let
        assert (1, 2, 3);
    tel;
}

fn main() {}
