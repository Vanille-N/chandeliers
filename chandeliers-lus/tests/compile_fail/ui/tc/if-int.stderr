error: The condition of if should be of type bool
 --> tests/compile_fail/ui/tc/if-int.rs:4:16
  |
4 |         assert if 1 then true else false;
  |                ^^^^^^^^^^^^^^^^^^^^^^^^^
  |
note: The argument is found to be of type int
 --> tests/compile_fail/ui/tc/if-int.rs:4:19
  |
4 |         assert if 1 then true else false;
  |                   ^

error: Expected a scalar type, found a tuple (int, int, int)
  --> tests/compile_fail/ui/tc/if-int.rs:11:19
   |
11 |         assert if (1, 2, 3) then true else false;
   |                   ^^^^^^^^^

error: The argument of assert should be of type bool
  --> tests/compile_fail/ui/tc/if-int.rs:18:9
   |
18 |         assert 1;
   |         ^^^^^^^^
   |
note: The argument is found to be of type int
  --> tests/compile_fail/ui/tc/if-int.rs:18:16
   |
18 |         assert 1;
   |                ^

error: Expected a scalar type, found a tuple (int, int, int)
  --> tests/compile_fail/ui/tc/if-int.rs:25:16
   |
25 |         assert (1, 2, 3);
   |                ^^^^^^^^^
